src/realm/group_writer.cpp

```diff
@@ -687,8 +687,12 @@ std::pair<size_t, size_t> GroupWriter::extend_free_space(size_t requested_size)
     // extended the file size. It can also happen as part of initial file expansion
     // during attach_file().
     size_t logical_file_size = to_size_t(m_group.m_top.get(2) / 2);
-    size_t extend_size = requested_size;
-    size_t new_file_size = logical_file_size + extend_size;
+    size_t new_file_size = logical_file_size;
+    if (REALM_UNLIKELY(int_add_with_overflow_detect(new_file_size, requested_size))) {
```

Comment:
Removing it relied on the assumption that we're going to run out of virtual address space before we get to a place where it could overflow. You're right, let's leave it in.